graspologic/pipeline/preconditions.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) Microsoft Corporation.
```

Comment:
is there a reason to keep any of these in pipeline, as opposed to root level module?

Reply:
1: I'm hoping we can get rid of the type checking after we check out whether beartype will work for us
2: yes, regardless of whether we get rid of the type checking part or not, it should go into a utils package.  Or at least be made available for the rest of graspologic.  I just wanted to set it as a trial balloon here to see if anyone would want to use it before I "graduated" it up in the package structure.  I can totally move it if you want me to now though

Reply:
i feel like i'm for that, even if we change the structure/details later